lib/internal/process/per_thread.js

```diff
@@ -255,6 +256,82 @@ const replaceUnderscoresRegex = /_/g;
 const leadingDashesRegex = /^--?/;
 const trailingValuesRegex = /=.*$/;
 
+class FlagsSet extends Set {
```

Comment:
If the the reason for putting the feature under `process` is code-sharing, this could be just split into a different file. I also think that `v8` is a more appropriate place for it.